File: 백준/Silver/11053. 가장 긴 증가하는 부분 수열/가장 긴 증가하는 부분 수열.js
const fs = require('fs')
const input = fs.readFileSync('/dev/stdin','utf8').trim().split('\n')

const N = Number(input.shift())
const values = input[0].trim().split(' ').map(Number)

const dp = new Array(N).fill(1)

for(let i=0;i<N;i++){
    for(let j=0;j<i;j++){
        if (values[i] > values[j]){
            dp[i] = Math.max(dp[i],dp[j]+1)
        }
    }
}

console.log(Math.max(...dp))

File: 백준/Silver/11053. 가장 긴 증가하는 부분 수열/가장 긴 증가하는 부분 수열.py
a = int(input())
arr = list(map(int,input().split()))

dp = [1]*a

for i in range(a):
  for j in range(0,i):
    if arr[i] > arr[j] :  dp[i] = max(dp[i],dp[j]+1)

print(max(dp))